documentation/topics/actions/actions.md

```diff
@@ -273,3 +273,383 @@ end
 ```elixir
 Ash.Changeset.for_create(Resource, :create, %{}, context: %{ip_address: "<ip_address>"})
 ```
+
+## Action Lifecycle
+
+This section provides a comprehensive overview of the Ash resource action lifecycle, detailing when each phase executes in relation to database transactions.
+
+### Overview
+
+Ash resource actions follow a well-defined lifecycle that ensures proper data validation, transformation, and persistence. The lifecycle is divided into three main phases:
+
+- **Pre-Transaction Phase** - Operations before database transaction
+- **Transaction Phase** - Operations within database transaction  
+- **Post-Transaction Phase** - Operations after database transaction
+
+#### Important Notes:
+
+- **Query Actions**: Read queries do not currently have `before_transaction`, `after_transaction`, or `around_transaction` callbacks
+- **Around Action Behavior**: `around_action` hooks do not complete their "end" phase if the action fails
```

Comment:
Lets add a note here that generic actions are left out because they currently do not support hooks of any kind, or preparations/changes/validations.